pkg/models/object.go

```diff
@@ -0,0 +1,62 @@
+package models
+
+// RawObject maps pretty closely to raw blob storage
+type RawObject struct {
+	UID  string `json:"uid"`
+	Kind string `json:"kind"`
+
+	// Who saved it when
+	Created   int64 `json:"created,omitempty"`   // timestamp
+	CreatedBy int64 `json:"createdBy,omitempty"` // user_id
+	Updated   int64 `json:"updated,omitempty"`   // timestamp  Last-Modified
+	UpdatedBy int64 `json:"updatedBy,omitempty"` // user_id
+
+	// Raw body
+	Size int64  `json:"size"`           // Content-Length
+	ETag string `json:"etag,omitempty"` // MD5 digest of the Body
+	Body []byte `json:"-"`              // don't return this as JSON unless explicit
+
+	// Optional additional properties managed by the underlying storage
+	Properties map[string]string `json:"properties,omitempty"`
+
+	// Not in every object store, but might be
+	Version     string `json:"version,omitempty"`  // commit hash, incrementing number (as string)
+	SyncTime    int64  `json:"syncTime,omitempty"` // when was the object synced from external source (provisioning or git)
+	SaveMessage string `json:"message,omitempty"`  // the commit message when this was saved
+}
+
+// ObjectSummary is derived from a RawObject and should not depend on system state
+type ObjectSummary struct {
+	Name        string            `json:"name,omitempty"`
+	Description string            `json:"description,omitempty"`
+	Labels      map[string]string `json:"labels,omitempty"` // "tags" are represented as keys with empty values
+	URL         string            `json:"URL,omitempty"`    // not great to save here, but maybe not terrible :shrug:
+	Error       string            `json:"error,omitempty"`  // Reason the object is invalid
+
+	// Optional values -- naming convention for types?
+	Fields map[string]interface{} `json:"fields,omitempty"` // Saved as JSON, returned in results, but values not sortable
```

Comment:
@sdboyer -- when this makes its way to an extendible structure, we will need a more strict way to define and produce what ends up in this field.  Pretty sure codegen + schemas will save the day

Reply:
i don't know what this is supposed to be - the object body?

Reply:
If so, maybe, but i don't know where instances of these objects are imagined being sent to and from, or in what sense the values are "optional" (like, in the schema sense, where the object being indexed may or may not contain a value - or perhaps from the perspective of the generic search system, in that they're additional vis-a-vis what the system actually expects?)

All i can see in looking at the test code is that `hasTemplateVars` is in one of the fixtures, which doesn't seem to be an actual dashboard field, so it would appear more like the latter - these are some wholly separate set of fields with an opaque relationship to the fields in the object schema itself, and therefore in need of a fully separate companion schema